bpdm-orchestrator/src/main/kotlin/org/eclipse/tractusx/bpdm/orchestrator/service/GoldenRecordTaskStorage.kt

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.orchestrator.service
+
+import org.eclipse.tractusx.bpdm.orchestrator.exception.BpdmDuplicateTaskIdException
+import org.eclipse.tractusx.bpdm.orchestrator.model.GoldenRecordTask
+import org.eclipse.tractusx.orchestrator.api.model.ResultState
+import org.eclipse.tractusx.orchestrator.api.model.StepState
+import org.eclipse.tractusx.orchestrator.api.model.TaskStep
+import org.springframework.stereotype.Service
+
+@Service
+class GoldenRecordTaskStorage {
+
+    private val tasks: MutableList<GoldenRecordTask> = mutableListOf()
```

Comment:
This may lead to concurrency exceptions if multiple threads from different requests access the list at the same time. I think the long term solution is to store it in a database anyway, so I think we should just think of a very low effort solution right now. Simpliest solution that I can think of right now is to limit the request thread pool to 1 basically ensuring sequential processing of requests: [server.tomcat.max-connections](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#application-properties.server.server.tomcat.max-connections)

Reply:
Or we could synchronize (`@Synchronize`) all the methods in `GoldenRecordTaskService`?
It's basically the same but is a little more fine grained and explicit.
This shouldn't have much impact as these methods are expected to be very fast.

Reply:
Yea that should work as well for now